packages/sails-ng-common/src/validation/validators.ts

```diff
@@ -374,4 +374,71 @@ export const formValidatorsSharedDefinitions: FormValidatorDefinition[] = [
       };
     },
   },
+  {
+    class: "orcid",
+    message: "@validator-error-orcid",
+    create: (config) => {
+      const optionNameKey = "class";
+      const optionNameValue = formValidatorGetDefinitionString(config, optionNameKey, "orcid");
+      const optionMessageKey = "message";
+      const optionMessageValue = formValidatorGetDefinitionString(config, optionMessageKey, "@validator-error-orcid");
+
+      return (control) => {
+        if (control.value == null || control.value === "") {
+          return null; // don't validate empty values
+        }
+
+        let value = control.value.toString();
+
+        // Validate format: either xxxx-xxxx-xxxx-xxxx or xxxxxxxxxxxxxxxx
+        if (!/^(\d{4}-\d{4}-\d{4}-\d{3}[\dX]|\d{15}[\dX])$/i.test(value)) {
+          return {
+            [optionNameValue]: {
+              [optionMessageKey]: optionMessageValue,
+              params: {
+                actual: control.value,
+              },
+            },
+          };
+        }
+
+        value = value.replace(/-/g, '');
+
+        if (value.length !== 16 || !/^\d{15}[\dX]$/i.test(value)) {
+          return {
+            [optionNameValue]: {
+              [optionMessageKey]: optionMessageValue,
+              params: {
+                actual: control.value,
+              },
+            },
+          };
+        }
+
+        const baseDigits = value.substring(0, 15);
+        const checkDigit = value.substring(15);
+
+        let total = 0;
```

Comment:
I suggest adding a comment with a link to the spec for the ORCID checksum algorithm.